cmd/allocator/main.go

```diff
@@ -241,18 +241,18 @@ func newServiceHandler(kubeClient kubernetes.Interface, agonesClient versioned.I
 		remoteAllocationTimeout,
 		totalRemoteAllocationTimeout)
 
-	stop := signals.NewStopChannel()
+	ctx := signals.NewSigKillContext()
 	h := serviceHandler{
 		allocationCallback: func(gsa *allocationv1.GameServerAllocation) (k8sruntime.Object, error) {
-			return allocator.Allocate(gsa, stop)
+			return allocator.Allocate(ctx, gsa)
 		},
 		mTLSDisabled: mTLSDisabled,
 		tlsDisabled:  tlsDisabled,
 	}
 
-	kubeInformerFactory.Start(stop)
-	agonesInformerFactory.Start(stop)
-	if err := allocator.Start(stop); err != nil {
+	kubeInformerFactory.Start(ctx.Done())
+	agonesInformerFactory.Start(ctx.Done())
+	if err := allocator.Start(ctx); err != nil {
```

Comment:
Because the allocator has to go all the way down to client-go to allocate a GameServer, we have to pass the ctx all the way through. 

And it also means the API for all systems is the same all the way through.

It's also possible that `allocation.Start(...)` should be changed to `allocation.Run(...)` to match all the other modules in Agones, hence the weirdness in the API surface, so I totally get the comment.